plugins/cryptolib/cryptolib.cpp

```diff
@@ -411,6 +411,52 @@ void verifyPKIAlgorithm(const char * pkAlgorithm)
 }
 
 
+//---------------------------------------------------------
+// Helper that reads the given logical file into a string buffer
+// Throws if user does not have read permission or if FNF
+//---------------------------------------------------------
+void loadLFS(const char * lfs, IUserDescriptor * user, StringBuffer &sb)
+{
+    CDfsLogicalFileName lfn;
+    lfn.set(lfs);
+    try
+    {
+        Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn, user, false, false, false, nullptr, defaultPrivilegedUser);//scope checks
+        if (!df)
+        {
+            throw MakeStringException(-1, "File Not Found");
+        }
+
+        unsigned numParts = df->numParts();
+        if (numParts != 1)
+        {
+            throw MakeStringException(-1, "Part count (%d) is incorrect", numParts);
```

Comment:
minor: would be good to mention the lfn in all these exception message.

Reply:
A while back in another review RKC didn't want me to display the (private) file spec because anyone with logfile access could find its location.  Perhaps not an issue so much now, since it is a LFN and might be scope protected.  Will do as you requested